game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1360,10 +1316,16 @@ private void fireNavalBombardment(final IDelegateBridge bridge) {
       }
       final List<Unit> allEnemyUnitsAliveOrWaitingToDie = new ArrayList<>(defendingUnits);
       allEnemyUnitsAliveOrWaitingToDie.addAll(defendingWaitingToDie);
+      final List<FiringGroup> firingGroups =
+          Bombard.builder()
+              .firingUnits(bombard)
+              .attackableUnits(attacked)
+              .defending(false)
+              .build()
+              .getFiringGroups();
```

Comment:
Is there a place where build() isn't immediately followed by getFiringGroups()? If not, should we just make build() return them? Or make buildFiringGroups()?

Reply:
@DanVanAtta had left a comment in the Battle Tree Estimator pull request that said that the code base had this pattern for utility classes.  See his comment at https://github.com/triplea-game/triplea/pull/6501#issuecomment-632436501.

Reply:
Indeed. The 'swingAction' variant I thought at the time was pretty nice. Thinking back to it, I just don't know which of these two is better:
```
UtilClass.methodCall(Parameters.builder()
    .param1(...)
    .param2(...)
    .build());
```

or:
```
UtilClass.builder()
    .param1(...)
    .param2(...)
    .build()
    .voidMethodCall();
```

This might be worth a code convention issue to discuss when/where to use either or if to prefer one. I can see the first being more useful if there are multiple methods that could be invoked, the latter is a more typical parameter object pattern. I'd guess that the latter is more idiomatic.

Reply:
Should I change anything?  Use a Parameters builder?